tests/event_loop_test.c

```diff
@@ -31,7 +32,7 @@ struct task_args {
     enum aws_task_status status;
     struct aws_mutex mutex;
     struct aws_condition_variable condition_variable;
-    bool thread_complete;
+    struct aws_atomic_var thread_complete;
```

Comment:
Should be atomic since it is written-by the cleanup thread and read-by the main thread.

Reply:
The data-race / ordering violation is:
- cleanup thread launched and detached
- main thread exits and frees allocator
- cleanup thread tries to use allocator to free exit_callback_data